docs/plugin-architecture.md

```diff
@@ -0,0 +1,307 @@
+# Plugin Architecture Documentation
+
+This document provides a comprehensive overview of the WooCommerce PayPal Payments plugin architecture, explaining its modular design and how the various components work together.
+
+## Overview
+
+The WooCommerce PayPal Payments plugin is built using a modular architecture powered by the **Inpsyde Modularity framework**. This design provides:
```

Comment:
Maybe use "Syde" instead of "Inpsyde"
```suggestion
The WooCommerce PayPal Payments plugin is built using a modular architecture powered by the **Syde Modularity framework**. This design provides:
```